### ISSUE TITLE
Generated zip packages are occasionally broken

### ISSUE THREAD

**Issue by digitarald** [mozilla/r2d2b2g#39]
Updating an app directory often results in a broken zip package.

Result: Installed app has name, but no icon; launching shows gray screen.

**Comment by mykmelez** [comment]
It'd be handy to have working and broken archives of the same app so we can compare them to figure out what the difference is between them.

**Comment by digitarald** [comment]
Working: http://dl.dropbox.com/u/1864007/Mozilla/issues/application.zip
Broken: http://dl.dropbox.com/u/1864007/Mozilla/issues/application-broken.zip

**Comment by mykmelez** [comment]
I bet it's simply that I forgot to close the ZIP file after writing to it.

http://mxr.mozilla.org/mozilla-central/source/modules/libjar/zipwriter/public/nsIZipWriter.idl#201

```
   addDirToArchive(writer, dir, "");
+
+  writer.close();

   console.log("archived dir " + dirToArchive);
```

**Comment by digitarald** [comment]
Fixed in #38
